host_vars/local.yaml

```diff
@@ -39,7 +39,7 @@ spark_version: 3.5.2
 spark_custom_name: hadoop3
 
 flink_enabled: false
-flink_binary_version: 1.20
+flink_binary_version: '1.20'
```

Comment:
interesting issue, ideally, we can evaluate it from the "flink_version"

Reply:
Thank you for the reminder. I tried it myself, and if I use the following code, it works successfully on my local computer, so we don't need the `FLINK_BINARY_VERSION`:

```
{% if flink_enabled and iceberg_enabled %}
FLINK_BINARY_VERSION=${FLINK_VERSION%.*}
ICEBERG_FLINK_JAR_NAME=iceberg-flink-runtime-${FLINK_BINARY_VERSION}
download_if_not_exists ${MAVEN_MIRROR}/org/apache/iceberg/${ICEBERG_FLINK_JAR_NAME}/${ICEBERG_VERSION}/${ICEBERG_FLINK_JAR_NAME}-${ICEBERG_VERSION}.jar
{% endif %}
```

However, I'm not sure if this is universally applicable or if it might introduce other issues. If it doesn't introduce any other problems, I'll push my local changes and directly remove the `FLINK_BINARY_VERSION` parameter.

Reply:
I find that we still need the `FLINK_BINARY_VERSION` parameter, especially during the `download` and `build-image` processes. In some custom scenarios, `FLINK_BINARY_VERSION` and `FLINK_VERSION` might differ, so keeping them separately specified currently seems to be the simpler approach.

Reply:
I mean something like below in `host_vars/local.yaml`

```
flink_version: 1.20.0
flink_binary_version: {{ substring of ${flink_version} from begin until the 2nd dot }}
```

and users are allowed to override it by 
```
ansible-playbook -e "{flink_binary_version: 1.19}"
```